research_project/greedy_triangulation.py

```diff
@@ -9,16 +9,32 @@
 
 from src.functions import poipairs_by_distance, new_edge_intersects
 
+def create_pois_groups(subgraph_percentages: list[float], pois: list[int]) -> list[int]:
+    pois_groups = []
+    pois_not_added = pois
```

Comment:
Copy the list, rather than referencing it
```suggestion
    pois_not_added = pois.copy()
```